File: Textilsoft/src/main/java/br/com/textilsoft/controller/ContaReceberController.java
package br.com.textilsoft.controller;

import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

import javax.ws.rs.Consumes;
import javax.ws.rs.DELETE;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import br.com.textilsoft.dao.ContaReceberDAO;
import br.com.textilsoft.model.ContaReceber;
import br.com.textilsoft.model.SituacaoContaReceber;


@Path("contasreceber")
public class ContaReceberController {

	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("/")
	public List<ContaReceber> listaContasPagar() {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			
			return ContaReceberDAO.listar();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("status/")
	public List<String> listarSituacao() {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			
			return ContaReceberDAO.listarStatus();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("statusobj/")
	public List<SituacaoContaReceber> listarSituacao2() {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			
			return ContaReceberDAO.listarStatusObj();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("{id}/")
	public ContaReceber getFornecedor(@PathParam("id") long id) {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			return ContaReceberDAO.selecionar(id);
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}

	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("status/{status}/")
	public List<ContaReceber> getFornecedor(@PathParam("status") String status) {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			
			return ContaReceberDAO.listarPorStatus(status);
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("pago/")
	public double pegarValorPago() {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			return ContaReceberDAO.pegarTotalPago();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("pendente/")
	public double pegarValorPendente(){
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			return ContaReceberDAO.pegarTotalPendente();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	@Path("atrasado/")
	public double pegarValorAtrasado() {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			return ContaReceberDAO.pegarTotalAtrasado();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	@POST
	@Consumes(MediaType.APPLICATION_JSON)
	@Path("/")
	public Response create(ContaReceber contaReceber) {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			ContaReceberDAO.inserir(contaReceber);
			
			return Response.status(Response.Status.CREATED).build();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@DELETE
	@Path("deletar/{id}/")
	public Response deletetotal(@PathParam("id") long id) {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			ContaReceberDAO.excluirtotal(id);
			return Response.status(Response.Status.OK).build();
		} catch (Exception ex) {
			Logger.getLogger(PedidoController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}
	
	@PUT
	@Consumes(MediaType.APPLICATION_JSON)
	@Path("/")
	public Response update(ContaReceber contaReceber) {
		try {

			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();			
			ContaReceberDAO.alterar(contaReceber);
		
			return Response.status(Response.Status.ACCEPTED).build();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}

	@DELETE
	@Path("{id}/")
	public Response delete(@PathParam("id") long id) {
		try {
			ContaReceberDAO ContaReceberDAO = new ContaReceberDAO();
			ContaReceberDAO.excluir(id);
			return Response.status(Response.Status.OK).build();
		} catch (Exception ex) {
			Logger.getLogger(ContaReceberController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}

}

File: Textilsoft/src/main/webapp/js/controller/conta-receber-controller.js
appTextilsoft.controller("contaReceberController", function($scope, $http,
		$routeParams) {
	$scope.listaContaReceber = [];
	$scope.contaReceber = {};
	$scope.contaReceber.Pago = 0;
	$scope.contaReceber.Pendente = 0;
	$scope.contaReceber.Atrasado = 0;	
	$scope.pago = 0;
	$scope.pendente = 0;
	$scope.atrasado = 0;
	$scope.excluir = 0;
	
	var url = 'http://localhost:8080/Textilsoft/rest/contasreceber/';
	

	
	$scope.listarStatus = function() {	
	
		$http({
			method : 'GET',
			url : url + 'statusobj/'
		}).then(function(response) {		
			
			$scope.contaReceber = response.data;	
			
			console.log($scope.contaReceber);
			$scope.contaReceber.filter(function(i, idx) {
			    if(i.status === "Pago") $scope.contaReceber.Pago = i.quantidade;
			    if(i.status === "Pendente") $scope.contaReceber.Pendente = i.quantidade;
			  
			    if(i.status === "Atrasado") $scope.contaReceber.Atrasado = i.quantidade;
			});						
		
				
			
		}, function(response) {

			console.log(response.data);
			console.log($scope.contaReceber);

		});
	};
	
	$scope.pegarValorPago = function() {		
		$http({
			method : 'GET',
			url : url + 'pago/'
		}).then(function(response) {		
			
			$scope.pago = response.data;		
			
		}, function(response) {

			console.log(response.data);
	

		});
	};
	$scope.pegarValorPendente = function() {		
		$http({
			method : 'GET',
			url : url + 'pendente/'
		}).then(function(response) {		
			
			$scope.pendente = response.data;
			
			
		}, function(response) {

			console.log(response.data);
	

		});
	};
	$scope.pegarValorAtrasado = function() {		
		$http({
			method : 'GET',
			url : url + 'atrasado/'
		}).then(function(response) {		
			
			$scope.atrasado = response.data;
		
			
		}, function(response) {

			console.log(response.data);
	

		});
	};
		
	$scope.listarPorStatus = function(status) {		
		$http({
			method : 'GET',
			url : url + 'status/' + status
		}).then(function(response) {		
			
			$scope.listaContaReceber = response.data;
			
		}, function(response) {

			console.log(response.data);
			console.log($scope.contaReceber);

		});
	};
	
	$scope.chamarMetodos = function(){
		
			
			$scope.listarPorStatus("Atrasado");
			$scope.pegarValorPago();
			$scope.pegarValorPendente();
			$scope.pegarValorAtrasado();
			$scope.listarStatus();
		}	
	
	$scope.listarValores = function(){
		
		$scope.pegarValorPago();
		$scope.pegarValorPendente();
		$scope.pegarValorAtrasado();
		$scope.listarStatus();
	}	
	
	$scope.salvarID = function(id){
			$scope.excluir = id;		
	}
	
		
	$scope.deletar = function(id) {

			$http({
				method : 'DELETE',
				url : url + id
			}).then(function(response) {			
				var pos = 0;
				$scope.listaContaReceber.filter(function(i, idx) {
				    if(i.idContaReceber == id)
				    	pos = idx; 				   
				});				
				 $scope.listaContaReceber.splice(pos, 1);			
				 $scope.listarValores();
				
			}, function(response) {
				console.log('error do salvar');
				console.log(response.data);
				console.log(response.status);
			});
	};
});


	